### PR TITLE
os/link

### PR DESCRIPTION
**Pull request by leongross**
Add `os.Link` functionality to create linux `hardlinks`

### CODE REVIEW

**Comment by deadprogram** [comment]
@rminnich any comment here before merge? :smile_cat:

File: src/os/os_hardlink_test.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
+//go:build !windows && !baremetal && !js && !wasi && !wasip1
```
Comment (deadprogram) [inline src/os/os_hardlink_test.go]: Do we also need to have `!wasm_unknown` in this list?

Comment (leongross) [reply src/os/os_hardlink_test.go]: this sounds like a reasonable thing to do

**Comment by deadprogram** [comment]
Thanks for adding this @leongross now merging.

**Comment by leongross** [comment]
Thank you for taking care of that!

**Pull request merged**